docs/core/testing/mstest-analyzers/overview.md

```diff
@@ -15,6 +15,42 @@ ms.date: 12/20/2023
 
 The rules are organized into categories such as performance usage...
 
+Starting with MSTest.TestFramework 3.7, the MSTest.Analyzers NuGet package is a dependency of the framework. For earlier versions, you need to use `MSTest` metapackage or add a package reference for `MSTest.Analyzers` explicitly.
+
+## MSTestAnalysisMode
+
+Starting with MSTest 3.8, an MSBuild property named `MSTestAnalysisMode` is available to determine which analyzers are enabled at which severity.
+
+The available values are:
+
+### `None`
+
+This sets all analyzers to `none` severity, making all of them disabled. You can then enable individual analyzers using `.editorconfig` or `.globalconfig` files.
+
+### `Default`
+
+This follows the default documented behavior for each rule.
+
+- Rules that are enabled by default will use their default severity.
+- Rules that are disabled by default will use `none` severity.
+
+### `Recommended`
+
+This mode is more aggressive than `Default`. Rules that are enabled by default with Info (`suggestion`) severity are escalated to be warnings. Moreover, certain rules may decide that they are escalated to be errors in both `Recommended` and `All` modes. For example, [MSTEST0003: Test methods should have valid layout](mstest0003.md) is escalated to error in `Recommended` and `All` modes.
```

Comment:
The first sentence could feel a little scary to users. I would instead say that in `Default` we are ensuring to not have rules that would cause warning or error and that don't result in runtime error. Recommended is what we think most dev should use but this means that a minor update of the package could cause new warnings.

Reply:
> Recommended is what we think most dev should use but this means that a minor update of the package could cause new warnings.

But a minor update can cause new warnings for "Default" as well?

Reply:
Only for cases where there is an execution error. Otherwise we mark the new rules as info